### DISCUSSION TITLE
Does Minio Java Client auto prefix when interacting with S3?

### DISCUSSION THREAD

**Discussion by dawickizer** [minio/minio#14343] [Q&A]
Hello,

So this code is successfully working for me when interacting with Minio being served on localhost:9001
```
  public void uploadFile(String objectName, MultipartFile file) throws IOException, NoSuchAlgorithmException, InvalidKeyException, MinioException {
    minioClient.putObject(
        PutObjectArgs.builder()
        .bucket(bucketName)
        .object(objectName)
        .stream(file.getInputStream(), file.getSize(), -1)
        .contentType(file.getContentType())
        .build());
  }
```
For example if `bucketName` is `my-bucket` and `objectName` is `my/path/to/file.jpg` and i run the above code...I can navigate to Minio and see that an object has been posted in bucket `my-bucket` with a path of `my/path/to/file.jpg`

However, when we run this code against our DEV environment its connecting to an actual s3 bucket from AWS, and this same exact code will show that a path of `my-bucket/my/path/to/file.jpg` is being created inside of `my-bucket`. So it ends up looking like `my-bucket/my-bucket/my/path/to/file.jpg`. It seems it is automatically adding an extra `my-bucket`. I have debugged everything to confirm that the `objectName` and `bucketName` are as expected so its not the code accidentally adding an extra `my-bucket`.

Not sure if I am doing something wrong, if this is a Minio bug, or if this is an AWS bug, or perhaps a fundamental difference in how AWS stores files vs how Minio stores files

**Comment by harshavardhana** [comment]
> However, when we run this code against our DEV environment its connecting to an actual s3 bucket from AWS, and this same exact code will show that a path of `my-bucket/my/path/to/file.jpg` is being created inside of `my-bucket`. So it ends up looking like `my-bucket/my-bucket/my/path/to/file.jpg`. It seems it is automatically adding an extra `my-bucket`. I have debugged everything to confirm that the `objectName` and `bucketName` are as expected so its not the code accidentally adding an extra `my-bucket`.

This looks like our minio-java issue can you open a github issue here https://github.com/minio/minio-java/issues with code snippets - on how you initialize the constructor when you talk to AWS ?
